File: packages/dart/lib/src/word/rand_finance.dart
import 'package:randino/src/types.dart';
import 'package:randino/src/word/rand_word.dart';

/// Money, and the words for what is done with it.
///
/// [randWord] with the theme already chosen; every other parameter is the same.
/// For the detail form, pass [WordTheme.finance] to `randWordDetails` — Dart has
/// no overloads, and seventeen more functions for it would be seventeen too many.
///
/// ```dart
/// randFinance(language: WordLanguage.ko, count: 3); // [이자, 환율, 장부]
/// randFinance(language: WordLanguage.en, count: 3); // [Ledger, Yield, Escrow]
/// ```
List<String> randFinance({
  WordLanguage? language,
  int count = 1,
  RandRealism realism = RandRealism.real,
  int? minLength,
  int? maxLength,
  String? startsWith,
  bool unique = false,
}) => randWord(
  language: language,
  theme: WordTheme.finance,
  count: count,
  realism: realism,
  minLength: minLength,
  maxLength: maxLength,
  startsWith: startsWith,
  unique: unique,
);

File: packages/javascript/lib/word/randFinance.ts
import type { RandThemedWordOptions, WordDetail } from '../_types/global.js';
import { themedWord } from './wordGenerator.js';

/**
 * Money, and the words for what is done with it.
 *
 * `randWord({ theme: 'finance' })` with the theme already chosen; every other
 * option is the same.
 *
 * @example
 * randFinance({ language: 'ko', count: 3 }); // ['이자', '환율', '장부']
 * randFinance({ language: 'en', count: 3 }); // ['Ledger', 'Yield', 'Escrow']
 */
export function randFinance(options?: RandThemedWordOptions & { output?: 'value' }): string[];
/**
 * Finance words with the language and theme each one came from.
 *
 * @example
 * randFinance({ language: 'ko', output: 'detail' });
 * // [{ word: '이자', language: 'ko', theme: 'finance' }]
 */
export function randFinance(options: RandThemedWordOptions & { output: 'detail' }): WordDetail[];
export function randFinance(options: RandThemedWordOptions = {}): string[] | WordDetail[] {
	return themedWord('finance', options);
}
